src/redux/rockets/rockets.js

```diff
@@ -40,10 +45,23 @@ export const fetchRocketsAPI = () => async (dispatch) => {
 // reducer
 const reducer = (state = initialState, action) => {
   switch (action.type) {
-    case ADD_RESERVATION:
-      return [...state, action.payload];
-    case REMOVE_RESERVATION:
-      return state.filter((book) => book.item_id !== action.payload);
+    case ADD_RESERVATION: {
+      const newState = state.map((rocket) => {
+        if (rocket.id !== action.payload.id) return rocket;
+
+        return { ...rocket, reservation: true };
+      });
+      return [...newState];
```

Comment:
Great getting the ID of the reserved rocket and updating the state. Returned new state object with all rockets 🙌